File: Software/OpenAstroTracker ASCOM/OpenAstroTracker ASCOM/SetupDialogForm.vb
Imports System.Windows.Forms
Imports System.Runtime.InteropServices
Imports ASCOM.Utilities
Imports ASCOM.OpenAstroTracker

<ComVisible(False)> _
Public Class SetupDialogForm

    Private Sub OK_Button_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles OK_Button.Click ' OK button event handler
        ' Persist new values of user settings to the ASCOM profile
        Telescope.comPort = ComboBoxComPort.SelectedItem ' Update the state variables with results from the dialogue
        Telescope.traceState = chkTrace.Checked
        Telescope.latitude = CDbl(txtLat.Text)
        Telescope.longitude = CDbl(txtLong.Text)
        Telescope.elevation = CInt(txtElevation.Text)
        Me.DialogResult = System.Windows.Forms.DialogResult.OK
        Me.Close()
    End Sub

    Private Sub Cancel_Button_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Cancel_Button.Click 'Cancel button event handler
        Me.DialogResult = System.Windows.Forms.DialogResult.Cancel
        Me.Close()
    End Sub

    Private Sub ShowAscomWebPage(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles PictureBox1.DoubleClick, PictureBox1.Click
        ' Click on ASCOM logo event handler
        Try
            System.Diagnostics.Process.Start("http://ascom-standards.org/")
        Catch noBrowser As System.ComponentModel.Win32Exception
            If noBrowser.ErrorCode = -2147467259 Then
                MessageBox.Show(noBrowser.Message)
            End If
        Catch other As System.Exception
            MessageBox.Show(other.Message)
        End Try
    End Sub

    Private Sub SetupDialogForm_Load(sender As System.Object, e As System.EventArgs) Handles MyBase.Load ' Form load event handler
        ' Retrieve current values of user settings from the ASCOM Profile
        InitUI()
    End Sub

    Private Sub InitUI()
        chkTrace.Checked = Telescope.traceState
        ' set the list of com ports to those that are currently available
        ComboBoxComPort.Items.Clear()
        ComboBoxComPort.Items.AddRange(System.IO.Ports.SerialPort.GetPortNames())       ' use System.IO because it's static
        ' select the current port if possible...
        If ComboBoxComPort.Items.Contains(Telescope.comPort) Then
            ComboBoxComPort.SelectedItem = Telescope.comPort
        End If
        txtLat.Text = Telescope.latitude.ToString
        txtLong.Text = Telescope.longitude.ToString
        txtElevation.Text = Telescope.elevation.ToString
    End Sub

    Private Sub txtLat_KeyPress(sender As Object, e As KeyPressEventArgs) Handles txtLat.KeyPress
        If Not Char.IsDigit(e.KeyChar) And Not Char.IsControl(e.KeyChar) And Not e.KeyChar = "." And Not e.KeyChar = "," And Not e.KeyChar = "-" Then
            e.Handled = True
        End If
    End Sub

    Private Sub txtLong_KeyPress(sender As Object, e As KeyPressEventArgs) Handles txtLong.KeyPress
        If Not Char.IsDigit(e.KeyChar) And Not Char.IsControl(e.KeyChar) And Not e.KeyChar = "." And Not e.KeyChar = "," And Not e.KeyChar = "-" Then
            e.Handled = True
        End If
    End Sub

    Private Sub txtElevation_KeyPress(sender As Object, e As KeyPressEventArgs) Handles txtElevation.KeyPress
        If Not Char.IsDigit(e.KeyChar) And Not Char.IsControl(e.KeyChar) Then
            e.Handled = True
        End If
    End Sub


End Class

File: Software/OpenAstroTracker ASCOM/TelescopeDriver/SetupDialogForm.cs
using System;
using System.ComponentModel;
using System.Runtime.InteropServices;
using System.Windows.Forms;

namespace ASCOM.OpenAstroTracker {
    [ComVisible(false)] // Form not registered for COM!
    public partial class SetupDialogForm : Form {
        private ProfileData _profile;

        public SetupDialogForm(ProfileData profile) {
            _profile = profile;
            InitializeComponent();
        }

        private void OK_Button_Click(System.Object sender, System.EventArgs e) // OK button event handler
        {
            // Persist new values of user settings to the ASCOM profile
            _profile.ComPort =
                (string) ComboBoxComPort.SelectedItem; // Update the state variables with results from the dialogue
            _profile.TraceState = chkTrace.Checked;
            _profile.Latitude = System.Convert.ToDouble(txtLat.Text);
            _profile.Longitude = System.Convert.ToDouble(txtLong.Text);
            _profile.Elevation = System.Convert.ToInt32(txtElevation.Text);
            this.DialogResult = System.Windows.Forms.DialogResult.OK;
            this.Close();
        }

        public ProfileData GetProfileData() => _profile;

        private void Cancel_Button_Click(System.Object sender, System.EventArgs e) // Cancel button event handler
        {
            this.DialogResult = System.Windows.Forms.DialogResult.Cancel;
            this.Close();
        }

        private void ShowAscomWebPage(System.Object sender, System.EventArgs e) {
            // Click on ASCOM logo event handler
            try {
                System.Diagnostics.Process.Start("http://ascom-standards.org/");
            }
            catch (Win32Exception noBrowser) {
                if (noBrowser.ErrorCode == -2147467259)
                    MessageBox.Show(noBrowser.Message);
            }
            catch (Exception other) {
                MessageBox.Show(other.Message);
            }
        }

        private void SetupDialogForm_Load(System.Object sender, System.EventArgs e) // Form load event handler
        {
            // Retrieve current values of user settings from the ASCOM Profile
            InitUI();
        }

        private void InitUI() {
            chkTrace.Checked = _profile.TraceState;
            // set the list of com ports to those that are currently available
            ComboBoxComPort.Items.Clear();
            ComboBoxComPort.Items.AddRange(System.IO.Ports.SerialPort.GetPortNames()); // use System.IO because it's static
            // select the current port if possible...
            if (ComboBoxComPort.Items.Contains(_profile.ComPort))
                ComboBoxComPort.SelectedItem = _profile.ComPort;
            txtLat.Text = _profile.Latitude.ToString();
            txtLong.Text = _profile.Longitude.ToString();
            txtElevation.Text = _profile.Elevation.ToString();
        }

        private void txtLat_KeyPress(object sender, KeyPressEventArgs e) {
            if (!char.IsDigit(e.KeyChar) && !char.IsControl(e.KeyChar) && e.KeyChar != '.' && e.KeyChar != ',' &&
                e.KeyChar != '-')
                e.Handled = true;
        }

        private void txtLong_KeyPress(object sender, KeyPressEventArgs e) {
            if (!char.IsDigit(e.KeyChar) && !char.IsControl(e.KeyChar) && e.KeyChar != '.' && e.KeyChar != ',' &&
                e.KeyChar != '-')
                e.Handled = true;
        }

        private void txtElevation_KeyPress(object sender, KeyPressEventArgs e) {
            if (!char.IsDigit(e.KeyChar) && !char.IsControl(e.KeyChar))
                e.Handled = true;
        }
    }
}